tests/testSphPoint.py

```diff
@@ -0,0 +1,615 @@
+
+        This test prevents misuse of approximate floating-point
+        equality -- if `__eq__` is implemented using AFP, then this
+        test will fail for some value of `delta`. Testing multiple
+        values is recommended.
+
+        Parameters
+        ----------
+        delta : `number`
+            The separation, in degrees, at which point equality may
+            become intransitive.
+        """
+        point1 = SphPoint((45.0 - delta)*degrees, -18.0*degrees);
+        point2 = SphPoint( 45.0         *degrees, -18.0*degrees);
+        point3 = SphPoint((45.0 + delta)*degrees, -18.0*degrees);
+
+        self.assertTrue(point1 != point2 or point2 != point3 or point1 == point3)
+        self.assertTrue(point3 != point1 or point1 != point2 or point3 == point2)
+        self.assertTrue(point2 == point3 or point3 != point1 or point2 == point1)
+
+    def testEqualityAlias(self):
+        """Test if == handles coordinate degeneracies correctly.
+        """
+        # Coordinate wrapping
+        self.assertEqual(SphPoint(360.0*degrees, -42.0*degrees),
+                         SphPoint(  0.0*degrees, -42.0*degrees))
+        self.assertEqual(SphPoint(-90.0*degrees, -42.0*degrees),
+                         SphPoint(270.0*degrees, -42.0*degrees))
+
+        # Polar degeneracy
+        self.assertEqual(SphPoint( 42.0*degrees,  90.0*degrees),
+                         SphPoint(270.0*degrees,  90.0*degrees))
+        self.assertEqual(SphPoint(-42.0*degrees, -90.0*degrees),
+                         SphPoint( 83.0*degrees, -90.0*degrees))
+
+        self.assertNotEqual(SphPoint(-42.0*degrees,  90.0*degrees),
+                            SphPoint( 83.0*degrees, -90.0*degrees))
+
+    def testInquality(self):
+        """Test if == and != give mutually consistent results.
+        """
+        for lon1, lat1 in self._dataset[0:20]:
+            point1 = SphPoint(lon1*degrees, lat1*degrees)
+            for lon2, lat2 in self._dataset[0:20]:
+                point2 = SphPoint(lon2*degrees, lat2*degrees)
+                self.assertTrue((point1 == point2) == (point2 == point1))
+                self.assertTrue((point1 != point2) == (point2 != point1))
+                self.assertTrue((point1 == point2) != (point1 != point2))
+
+    def testBearingToError(self):
+        """Test if bearingTo() correctly handles invalid input.
+        """
+        destination = SphPoint(42.0*degrees, -90.0*degrees)
+
+        northPole = SphPoint(0.0*degrees,  90.0*degrees)
+        southPole = SphPoint(0.0*degrees, -90.0*degrees)
+        safe = SphPoint(0.0*degrees, 0.0*degrees)
+
+        with self.assertRaises(pexEx.DomainError):
+            northPole.bearingTo(destination)
+        with self.assertRaises(pexEx.DomainError):
+            southPole.bearingTo(destination)
+        with self.assertRaises(pexEx.DomainError):
+            destination.bearingTo(safe)
+
+        # Argument is allowed to be a pole.
+        safe.bearingTo(destination)
```

Comment:
I suggest ditching this because it duplicates `testBearingToValue`, which is more rigorous.